### PR TITLE
Transition checkpoint_dir to root_dir (part of the New CLI transition)

### PR DESCRIPTION
**Pull request by rasbt**

### CODE REVIEW

File: litgpt/config.py [hunk 1]
```diff
@@ -91,6 +91,16 @@ def __post_init__(self):
 
         self.rope_n_elem = int(self.rotary_percentage * self.head_size)
 
+    @classmethod
+    def get_name_from_hf_config(cls, hf_name: str) -> str:
```
Comment (awaelchli) [inline litgpt/config.py:95]: 1) How about implementing this in `from_name`? I don't see the need to duplicate the functionality. 
2) The class is not used, so this isn't really a classmethod

File: litgpt/pretrain.py [hunk 2]
```diff
@@ -83,20 +82,39 @@ def setup(
...
+                available_models = "\n".join(sorted(name_to_config))
+                print(f"Available values:\n{available_models}")
+                quit()
+    else:
+        model_name = "placeholder"
+
     hparams = capture_hparams()
     data = TinyLlama() if data is None else data
-    if model_config is not None and model_name is not None:
-        raise ValueError("Only one of `model_name` or `model_config` can be set.")
```
Comment (awaelchli) [inline litgpt/pretrain.py:96]: How do we provide a config now? I think now you'd have to set both model_name and config. Can you update the pretraining config files?

File: litgpt/config.py (hunk 1 above)
Comment (rasbt) [reply litgpt/config.py:95]: Thanks, I merged these two

File: litgpt/pretrain.py [hunk 3]
```diff
@@ -83,20 +82,39 @@ def setup(
...
+        num_slashes = model_name.count("/")
+        if num_slashes > 1:
+            raise ValueError(
+                f"Invalid model_name: {model_name}. The model_name argument cannot contain multiple slashes."
+                " It should be either in the format 'meta-llama/Meta-Llama-3-8B' or 'Meta-Llama-3-8B'."
+                )
+        if num_slashes:
+            try:
+                model_name = Config.get_name_from_hf_config(model_name)
+            except ValueError:
```
[the remark this answers is not in the document]
Comment (rasbt) [reply litgpt/pretrain.py:108]: Addressed it!

File: litgpt/pretrain.py (hunk 2 above)
Comment (rasbt) [reply litgpt/pretrain.py:96]: Good point, should be addressed.

**Pull request merged**